File: IKeep.Lib/Models/Relations/ElementGenericTask.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace IKeep.Lib.Models
{
    public class ElementGenericTask : Entity
    {
        public Guid ElementId { get; set; }
        [JsonIgnore]
        public virtual Element Element { get; set; }

        public string ElementName
        {
            get
            {
                return Element == null ? "" : Element.Name;
            }
        }
        public Guid GenericTaskId { get; set; }
        [JsonIgnore]
        public virtual GenericTask GenericTask { get; set; }
        public string GenericTaskDescription
        {
            get
            {
                return GenericTask == null ? "" : GenericTask.Description;
            }
        }
        public Status Status { get; set; }
    }

    //public enum Status
    //{
    //    Inactive,
    //    Active
    //}
}

File: IKeep.Web/ClientApp/src/models/ElementGenericTask.js
import Entity from './Entity'

class ElementGenericTask extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.ElementId = json.elementId;
            this.GenericTaskId = json.genericTaskId;
            this.Status = json.status;
            this.ElementName = json.generiElementName;
            this.GenericTaskDescription = json.genericTaskDescription;
        }
        else
        {
            this.ElementId = "";
            this.GenericTaskId = "";
            this.Status = 1;
            this.ElementName = "";
            this.GenericTaskDescription = "";
        }
        
    }
}

export default ElementGenericTask;